### DISCUSSION TITLE
GIN - additional MLP?

### DISCUSSION THREAD

**Discussion by vthost** [pyg-team/pytorch_geometric#4536] [Ideas]
Hi!
I just came across this and wanted to mention it in case you missed it so far as well :)
In the original paper by Xu et al., the general equation in Cor.6 considers an additional function f before the aggregation summation. In GIN, this MLP is dropped since it comes with the below assumption. (If I am not missing anything) I think GINConv drops the MLP as well, however  users may not use one-hot features. Maybe you can add the assumption in the comments or and additional version w/ a second MLP?
```
In the ﬁrst iteration, we do not need MLPs before summation if input features are one-hot encodings as their summation alone is injective. 
```

**Comment by rusty1s** [comment]
That is correct. We follow the final `GINConv` definition here. If your features are not one-hot encoded, and you want to ensure maximal expressivity, you have to apply an `MLP` before-hand:
```python
x = self.mlp(x)
x = self.conv1(x, edge_index)
...
```

**Reply by vthost** [reply]
Thank you for confirming!
